src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -119,17 +126,41 @@ public CheckResult Check(string expressionText, RecordType parameterType = null,
         public CheckResult Check(ParseResult parse, RecordType parameterType = null, ParserOptions options = null)
         {
             var bindingConfig = new BindingConfig(allowsSideEffects: options?.AllowsSideEffects == true);
-            return CheckInternal(parse, parameterType, bindingConfig);
+            var symbols = SymbolTable.NewFromRecord(parameterType);
+
+            return Check2(parse, bindingConfig, symbols);
+        }
+
+        public CheckResult Check2(
+            string expressionText,
+            ParserOptions parserOptions = null,
+            SymbolTable symbolTable = null)
```

Comment:
Should we have a class instead of 2 params
This could easily be extended w/o having a breaking change